File: 1399_count_largest_group.cpp
class Solution {
public:
    int countLargestGroup(int z) {
        int x[37]{}, m = 0, c = 0;
        for (int i = 1; i <= z; ++i) m = max(m, ++x[acc(i)]);
        for (int i : x) if (i == m) ++c;
        return c;
    }
    int acc(int v) { int s = 0; while (v) s += v % 10, v /= 10; return s; }
};

File: 1399_count_largest_group.java
class Solution {
    public int countLargestGroup(int n) {
        int[] a = new int[37];
        int max = 0, res = 0;
        for (int i = 1; i <= n; i++) {
            int s = 0, x = i;
            while (x > 0) { s += x % 10; x /= 10; }
            max = Math.max(max, ++a[s]);
        }
        for (int v : a) if (v == max) res++;
        return res;
    }
}
